File: be-fschool/src/main/java/com/tinder/befschool/entity/Conduct.java
package com.tinder.befschool.entity;

import jakarta.persistence.*;

@Entity
@Table(name = "conduct")
public class Conduct extends Auditable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "student_id", nullable = false)
    private Long studentId;

    @Column(name = "student_name")
    private String studentName;

    @Column(name = "teacher_id")
    private Long teacherId;

    @Column(name = "teacher_name")
    private String teacherName;

    @Column(name = "class_name")
    private String className;

    private Integer month;

    private Integer year;

    // EXCELLENT, GOOD, AVERAGE, WEAK
    @Column(name = "conduct_rating")
    private String conductRating;

    @Column(columnDefinition = "TEXT")
    private String comment;

    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }

    public Long getStudentId() { return studentId; }
    public void setStudentId(Long studentId) { this.studentId = studentId; }

    public String getStudentName() { return studentName; }
    public void setStudentName(String studentName) { this.studentName = studentName; }

    public Long getTeacherId() { return teacherId; }
    public void setTeacherId(Long teacherId) { this.teacherId = teacherId; }

    public String getTeacherName() { return teacherName; }
    public void setTeacherName(String teacherName) { this.teacherName = teacherName; }

    public String getClassName() { return className; }
    public void setClassName(String className) { this.className = className; }

    public Integer getMonth() { return month; }
    public void setMonth(Integer month) { this.month = month; }

    public Integer getYear() { return year; }
    public void setYear(Integer year) { this.year = year; }

    public String getConductRating() { return conductRating; }
    public void setConductRating(String conductRating) { this.conductRating = conductRating; }

    public String getComment() { return comment; }
    public void setComment(String comment) { this.comment = comment; }
}

File: prm/lib/models/conduct.dart
class Conduct {
  final int id;
  final int studentId;
  final String? studentName;
  final int? teacherId;
  final String? teacherName;
  final String? className;
  final int month;
  final int year;
  final String conductRating; // EXCELLENT, GOOD, AVERAGE, WEAK
  final String? comment;

  const Conduct({
    required this.id,
    required this.studentId,
    this.studentName,
    this.teacherId,
    this.teacherName,
    this.className,
    required this.month,
    required this.year,
    required this.conductRating,
    this.comment,
  });

  String get ratingLabel {
    switch (conductRating) {
      case 'EXCELLENT':
        return 'Xuất sắc';
      case 'GOOD':
        return 'Tốt';
      case 'AVERAGE':
        return 'Khá';
      case 'WEAK':
        return 'Yếu';
      default:
        return conductRating;
    }
  }

  factory Conduct.fromJson(Map<String, dynamic> json) {
    return Conduct(
      id: json['id'] ?? 0,
      studentId: json['studentId'] ?? 0,
      studentName: json['studentName']?.toString(),
      teacherId: json['teacherId'],
      teacherName: json['teacherName']?.toString(),
      className: json['className']?.toString(),
      month: json['month'] ?? 0,
      year: json['year'] ?? 0,
      conductRating: json['conductRating']?.toString() ?? 'GOOD',
      comment: json['comment']?.toString(),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'studentId': studentId,
      'studentName': studentName,
      'teacherId': teacherId,
      'teacherName': teacherName,
      'className': className,
      'month': month,
      'year': year,
      'conductRating': conductRating,
      'comment': comment,
    };
  }
}
